vscripts/confogl.nut

```diff
@@ -2,65 +2,308 @@
 MutationOptions <-
 {
   cm_ProhibitBosses = 0
- 	cm_AllowPillConversion = 0
+  cm_AllowPillConversion = 0
 }
 
-
 local pluginScopes = []
-Hooks <- {
-  function AllowBash(basher, bashee) {
-    // Default to allow the bash.
-    local retVal = true;
-	  for ( local idx = pluginScopes.len(); idx >= 0; --idx ) {
-      local nextPlugin = pluginScopes[idx];
-      local fn = pluginScopes[idx].AllowBash;
-      if ( typeof (fn) == "function") {
-        local res = fn(basher, bashee)
-        // If any plugin returns false, we return false.
-        if(res == false) {
-          return false;
-        }
-      }
+
+// int AllowBash(handle basher, handle bashee): called whenever melee bash is used.
+// Returns these values:
+// ALLOW_BASH_ALL - normal melee behavior
+// ALLOW_BASH_NONE - do nothing at all
+// ALLOW_BASH_PUSHONLY - applies physics push but deals no damage (including prevention of insta-kill ambush behavior)
+function AllowBash(basher, bashee) {
+  Msg(format("AllowBash(%s, %s)\n", basher.tostring(), bashee.tostring()));
+  // Default to allow the bash.
+  local retVal = ALLOW_BASH_ALL;
+  for ( local idx = pluginScopes.len() - 1; idx >= 0; --idx ) {
```

Comment:
id if we really want to run in reverse order, but maybe we do. :shrug:  I just had this copy pasted from valve's code.

Reply:
It makes logical sense to me that the last loaded plugin that overrides the function is the first one checked. I suppose that "it makes sense to me" is pretty arbitrary though.

Reply:
Now that we're using runPluginHookCallbacks(), I suppose it's actually the first plugin loaded that overrides the function that gets to dictate the return value now.